tests/foreman/ui/test_gpgkey.py

```diff
@@ -384,7 +380,118 @@ def test_consume_content_1(self):
 
         """
 
-        pass
+        product_name = gen_string("alpha", 8)
+        repository_name = gen_string("alpha", 8)
+        activation_key_name = gen_string("alpha", 8)
+        key_content = read_data_file(ZOO_CUSTOM_GPG_KEY)
+        # step1: Create gpg-key
+        gpgkey_attrs = entities.GPGKey(
+            name=key_name,
+            content=key_content,
+            organization=self.org_id
+        ).create_json()
+        # step 1.2: Create new lifecycle environments
+        lifecycle_env = entities.LifecycleEnvironment(
+            organization=self.org_id
+        ).create_json()
+        # step2: Creates new product without selecting GPGkey
+        product_id = entities.Product(
+            name=product_name,
+            organization=self.org_id
+        ).create_json()['id']
+        # step3: Creates new repository with GPGKey
+        repo_id = entities.Repository(
+            name=repository_name,
+            url=FAKE_1_YUM_REPO,
+            product=product_id,
+            gpg_key=gpgkey_attrs['id'],
+        ).create_json()['id']
+        # step 3.1: sync repository
+        entities.Repository(id=repo_id).sync()
+        # step 4: Create content view
+        content_view = entities.ContentView(
+            organization=self.org_id
+        ).create_json()
+        # step 5: Associate repository to new content view
+        response = client.put(
+            entities.ContentView(id=content_view['id']).path(),
+            {u'repository_ids': [repo_id]},
+            auth=get_server_credentials(),
+            verify=False,
+        )
+        response.raise_for_status()
```

Comment:
As of [NailGun 0.7.0](https://github.com/SatelliteQE/nailgun/releases/tag/0.7.0), the `create` method has finally been implemented. As a result, you can write code like the following:

```python
repo = entities.Repository(…).create()
repo.sync()
content_view = entities.ContentView(organization=self.org_id).create()
client.put(
    content_view.path(),
    {u'repository_ids': [repo.id]},
    auth=get_server_credentials(),
    verify=False,
).raise_for_status()
```